unit/src/ParsedFunctionTest.C

```diff
@@ -34,10 +34,12 @@ ParsedFunctionTest::init()
 
   InputParameters mesh_params = _factory->getValidParams("GeneratedMesh");
   mesh_params.set<MooseEnum>("dim") = "3";
+  mesh_params.set<std::string>("name") = "mesh";
```

Comment:
We do automatically generate the name, we use `getShortName()` in the Action system, see `AddAuxKernelAction` for example.